CHANGELOG.md

```diff
@@ -2,13 +2,23 @@
 
 ### Breaking changes
 
-* `RealmResults.distinct()` returns a new `RealmResults` object instead of filtering on the original object.
+* `RealmResults.distinct()` returns a new `RealmResults` object instead of filtering on the original object (#2947).
+
+### Enhancements
+
+* Add support for sorting by link's field (#672).
 
 ### Object Server API Changes (In Beta)
 
 * Add a default `UserStore` based on the Realm Object Store (`ObjectStoreUserStore`).
 * Change the order of arguments to SyncCredentials.custom to match iOS: token, provider, userInfo
 
+### Internal
+
+* Use Object Store's `Results` as the backend for `RealmResults` (#3372).
+  - Use Object Store's notification mechanism to trigger listeners.
+  - Local commit triggers Realm global listener and `RealmObject` listener on current thread immediately instead of in the next event loop.
```

Comment:
@cmelchior oh I didn't know adding `RealmChangeListener` **after** a commit would also be called. That's weird! I figured if I wanted the change listener to be called, I'd add it before I do something. But yes, that is a behavioral change.

I think iterator behavior was mostly a concern due to live updating of RealmResults in transactions, but that's fixed by replacing them with a `snapshot()` for the duration of the transaction

But now that I think about it, badly written Realm code can break if you do something like this:

``` java
    // main thread
    try(Realm r = Realm.getDefaultInstance()) {
        for(Dog dog : r.where(Dog.class).greaterThanOrEqual("age", 12).findAll()) {
            r.executeTransaction((realm) -> { // UI thread transaction
                 dog.age = 11;
            });
        }
    } 
```